hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -319,7 +320,7 @@ case class HoodieFileIndex(spark: SparkSession,
     val prunedPartitionsTuple: (Boolean, Seq[PartitionPath]) =
       if (isPartitionedTable && partitionFilters.nonEmpty) {
         // For partitioned table and partition filters, prune the partitions by the partition filters
-        if (shouldEmbedFileSlices) {
+        if (shouldConvertFilter) {
```

Comment:
What is the rationale behind this change?

Reply:
In the datasource implementation, the filter conversion is done inside the relation. But for the hadoopfsfactory, it still needs to be called. shouldEmbedFileSlices was originally set to true whenever we used the fg reader, but now that we are more selective, it needs to be controlled by a separate variable.